revolv/administrator/views.py

```diff
@@ -29,6 +35,423 @@ def get_context_data(self, **kwargs):
+    project_filter = Project.objects.all()
+
+    project_dict = {}
+
+
+    """
+    A helper function that calculates the initial cash balance from the
+    beginning of time to self.start_date
+
+    :return:
+        Returns a number that represents initial cash balance.
+    """
+    def calculateCashBalanceFromStart(self):
+        # THESE ARE DUMMY VARIABLES - USE THE REVOLV EARNINGS MODEL INSTEAD
+        payment_service_fee_rate = 0.01
+        revolv_earnings_rate = 0.07
+        allDonations = sum([payment.amount for payment in Payment.objects.donations(end_date=self.start_date)])
+        allReinvestments = sum([payment.amount for payment in AdminReinvestment.objects.reinvestments(end_date=self.start_date)])
+        allRepayments = sum([payment.amount for payment in AdminRepayment.objects.repayments(end_date=self.start_date)])
+        cashInAdjustments = sum([payment.amount for payment in AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_in')])
+        cashOutAdjustments = sum([payment.amount for payment in AdminAdjustment.objects.adjustments(end_date=self.start_date, cash_type='cash_out')])
+
+        cash_in = (allDonations * (1 - payment_service_fee_rate)) + (allRepayments * (1 - revolv_earnings_rate)) + cashInAdjustments
+        cash_out  = cashOutAdjustments + allReinvestments
+        net = cash_in - cash_out
+        return net
+    
+    """
+    A helper function that cleans a project name to make it JQuery 
+    friendly for use in the Accounting page.
+
+    :args:
+        name: A string to be cleaned
+
+    :return:
+        Returns a cleaned up version of name.
+    """
+    def clean_project_name(self, name):
+        name = name.replace(' ', '-').replace(';', 'sm').replace(':', 'co').replace("'", 'ap').replace('!', 'ex')
+        return name
+    
+    def make_project_dict(self):
+        self.project_dict = {}
+        for proj in Project.objects.all():
+            name = proj.title
+            self.project_dict[self.clean_project_name(name)] = name
+    
+    """
+    Updates class variables such as start_date, end_date, and project_filter
+    based on the keys passed in.
+
+    :args:
+        keys: The request passed in to the AccountingJSONView
+    """
+    def updateFilterVariables(self, keys):
+        self.make_project_dict()
+
+        try:
+            if 'start_date' in keys:
+                self.start_date = datetime.datetime.strptime(keys['start_date'], '%Y-%m-%d')
+            elif 'end_date' in keys:
+                self.end_date = datetime.datetime.strptime(keys['end_date'], '%Y-%m-%d')
+            elif 'project_choice' in keys:
+                if keys['project_choice'] == 'all':
+                    self.project_filter = Project.objects.all()
+                else:
+                    real_name = self.project_dict[keys['project_choice']]
+                    self.project_filter = Project.objects.filter(title=real_name)
+        except ValueError:
```

Comment:
I feel like this deserves a comment saying when it would be raised and why it's safe to ignore it